docs/feature-flags.md

```diff
@@ -0,0 +1,31 @@
+# Feature flags and partial releases
+
+Feature flags are an important tool that enables [trunk-based development](https://trunkbaseddevelopment.com/). They allow in-progress features to be merged into the main branch while still allowing that branch to be deployed to production at any time, thus decoupling application deploys from feature releases. For a deeper introduction, [Martin Fowler's article on Feature Toggles](https://martinfowler.com/articles/feature-toggles.html) and [LaunchDarkly's blog post on feature flags](https://launchdarkly.com/blog/what-are-feature-flags/) are both great articles that explain the what and why of feature flags.
+
+## How it works
+
+This project leverages [Amazon CloudWatch Evidently](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Evidently.html) to create and manage feature flags.
+
+## Creating feature flags
+
+The list of feature flags for an application is defined in the `feature_flags` property in its app-config module (in /infra/[app_name]/app-config/main.tf). To create a new feature flag, add a new string to that list. To remove a feature flag, remove the feature flag from the list. The set of feature flags will be updated on the next terraform apply of the service layer, or during the next deploy.
```

Comment:
Do we want to call out here and in the code that this is for boolean flags only since there are other types?

Reply:
I personally don't think so. A project team only ever needs booleans, I don't much value to the other types. The other types start to look like business logic to me, which should be in the application code.

For split tests, since there may be more than one non-control variation, I think we'd use either a string (just the name of the variation) or an integer (the index of the variation) instead of boolean, but split tests are out of scope for this change.